can/interfaces/socketcan/socketcan.py

```diff
@@ -260,11 +321,24 @@ def build_bcm_update_header(can_id: int, msg_flags: int, nframes: int = 1) -> by
 
 
 def dissect_can_frame(frame: bytes) -> Tuple[int, int, int, bytes]:
-    can_id, can_dlc, flags = CAN_FRAME_HEADER_STRUCT.unpack_from(frame)
+    can_id, data_len, flags, len8_dlc = CAN_FRAME_HEADER_STRUCT.unpack_from(frame)
     if len(frame) != constants.CANFD_MTU:
         # Flags not valid in non-FD frames
         flags = 0
-    return can_id, can_dlc, flags, frame[8 : 8 + can_dlc]
+
+    if data_len not in can.util.CAN_FD_DLC:
```

Comment:
Is there a specific reason you don't trust the value returned by the kernel? I'm not saying it's a bad thing, just interested in why you added this.

Reply:
As discussed with @hartkopp socketcan allow to send non valid FD frames with length not supported by CAN FD. For example, DLC=0x9 (data_len=12) and actual data_len=10 - this is invalid CAN FD frame but still could be received from the kernel.
To handle this case I added code to round up the amount of bytes note that no read overflow will happen because the CANFD MTU is set to 72 (64 bytes of data) in the socketcan implementation so the additional bytes read from the frame will still be part of the struct and will probably contains 0x00s. If you prefer we can add padding with a defined value to be on the safe side.